File: 0268-missing-number/0268-missing-number.java
class Solution {
    public int missingNumber(int[] nums) {
        int org_xor = 0;
        int given_xor = 0;
        for(int i=1;i<=nums.length;++i)
            org_xor ^= i;
        for(int x : nums)
            given_xor ^= x;
        return org_xor^given_xor;
    }
}

File: 0268-missing-number/0268-missing-number.cpp
class Solution {
public:
    int missingNumber(vector<int>& nums) {
        int xorVal = 0;
        for (int i=0; i<=nums.size(); i++) xorVal ^= i;
        for (const int &x: nums) xorVal ^= x;
        return xorVal;
    }
};